src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -624,47 +619,16 @@ void GameObject::Update(uint32 diff)
                     if (GameTime::GetGameTimeMS() < m_cooldownTime)
                         break;
 
-                    // Type 2 (bomb) does not need to be triggered by a unit and despawns after casting its spell.
-                    if (goInfo->trap.type == 2)
-                    {
-                        SetLootState(GO_ACTIVATED);
+                    // If diameter is 0 -> trap should not be activated when you're near it... If it should, check for areatriggers.
+                    if (!goInfo->trap.radius)
                         break;
-                    }
-
-                    // Type 0 despawns after being triggered, type 1 does not.
-                    /// @todo This is activation radius. Casting radius must be selected from spell data.
-                    float radius;
-                    if (!goInfo->trap.diameter)
-                    {
-                        // Battleground traps: data2 == 0 && data5 == 3
-                        if (goInfo->trap.cooldown != 3)
-                            break;
-
-                        radius = 3.f;
-                    }
-                    else
-                        radius = goInfo->trap.diameter / 2.f;
 
                     // Pointer to appropriate target if found any
                     Unit* target = nullptr;
 
-                    /// @todo this hack with search required until GO casting not implemented
```

Comment:
This is a wrong change. Thanks to a new field in GameObjectTemplate in master branch, we know that traps permanently spawned only trigger for players and summoned traps for any unit

Reply:
Hmmm, the spirit alarms in ICC also trigger on hunter pets and eye of killrog. They might be summoned traps tho.
I noticed that while running around i didnt trigger a trap but then later at that same spot there was one.

So... Ignore that spirit alarm case and implement the check like it was before?